README.md

```diff
@@ -1,2 +1,105 @@
-# deployer
-Deployer base config to setup our shopware deployment
+# deployer for shopware
+
+This repository contains a [deployer](https://deployer.org/) configuration for shopware.
+
+## Preparations
+
+### Repository
+Copy the contents of this repository to your shop project. Within your shop project download the latest [composer.phar](https://getcomposer.org/download/) and [deployer.phar](https://deployer.org/download) to the ``bin`` directory.
+Also download the latest [cachetool.phar](https://gordalina.github.io/cachetool/) if you need to reset an OPCache.
+
+The `web` directory will be used as shopware root directory. Place all custom files for your shopware project here, e.g. plugins and custom themes.
+
+We recommend to use [K10rProject](https://github.com/kellerkinderDE/K10rProject) for database migrations and snippet management so this config depends on it. Please add it to your repository if you want to use it. However if you don't need it you will need to remove it from the plugin list in `deploy.php` and remove the command `k10r:snippets:update` in the task `kellerkinder:shopware:config`.
+
+## Configurations
+
+### deploy-hosts.yaml
+
+This file defines the different hosts you may want to deploy to. Adjust those hosts to your needs.
+See [deployer documentation](https://deployer.org/docs/hosts) for further information.
+
+### deploy.php
+
+This file defines the basic configuration via deployers `set`-method. Make sure to adjust the following options to your needs:
+* `repository`: URL to the repository you want to deploy.
+* `bin/php`: Absolute path to the php executable on your server.
+* `bin/composer`: Path to composer. Adjust if you do not want to use the composer.phar.
+* `shopware_download_path`: URL to the shopware installation package you want to deploy.
+* `shopware_update_version`: Shopware version you want to update to. Will be used to check wether the current installation needs to be updated.
+* `shopware_update_path`: URL to shopware update package you want to update to.
+* `plugins`: An array of plugins that will be installed and updated during deployment. Those plugins need to be present in the codebase.
+* `theme_config`: An array of theme configuration settings that will be set during deployment.
```

Comment:
Do you want to describe `plugin_config` and `shopware_config` here as well?

Reply:
Added :)